src/component.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable, IOnChangesObject } from 'angular';
+
+/**
+ * Valid properties to make an angularJS component. The `controller` and `controllerAs` properties
+ * have been removed since we are trying to wrap a component to make it look more like an Angular
+ * component.
+ */
+interface INgComponentOptions {
+  selector: string;
+  template?: string | Injectable<(...args: any[]) => string>;
+  templateUrl?: string | Injectable<(...args: any[]) => string>;
+  bindings?: {[boundProperty: string]: string};
+  transclude?: boolean | {[slot: string]: string};
+  require?: {[controller: string]: string};
+}
+
+/**
+ * Attaches `__component` to the clazz object. If it already has one, then it overrides the
+ * properties it needs to.
+ *
```

Comment:
I'm not sure I understand this comment. What overrides what here? Vague pronoun "it" makes it hard to know. Does the "attach" method override the properties existing on the clazz object?